java/google/registry/flows/FlowMetadata.java

```diff
@@ -0,0 +1,24 @@
+package google.registry.flows;
+
+import com.google.auto.value.AutoValue;
+import google.registry.model.ImmutableObject;
+
+/** Object to allow setting and retrieving metadata information in flows. */
+@AutoValue
+public abstract class FlowMetadata extends ImmutableObject {
+
+  public abstract boolean superuser();
```

Comment:
@nfelt I didn't add a javadoc to this as, like you state, it's self-evident. [Also the existing code is inconsistent.](https://github.com/google/nomulus/blob/master/java/google/registry/flows/custom/DomainCheckFlowCustomLogic.java#L102) However I'd be happy to add the doc.

@CydeWeys naming the variable 'isSuperuser' was inconsistent with the rest of the parameter objects. Methods like [historyEntry](https://github.com/google/nomulus/blob/master/java/google/registry/flows/custom/DomainCreateFlowCustomLogic.java#L135), [entityChanges](https://github.com/google/nomulus/blob/master/java/google/registry/flows/custom/DomainCreateFlowCustomLogic.java#L146), or [years](https://github.com/google/nomulus/blob/master/java/google/registry/flows/custom/DomainCreateFlowCustomLogic.java#L153) would need to become getHistoryEntry, getEntityChanges, getYears for isSuperuser to match. As this is not the current case I named it to follow suit. However if 'isSuperuser' is preferred I'll make the change.